pydax/_dataset.py

```diff
@@ -119,17 +119,32 @@ def _pydax_dir(self) -> pathlib.Path:
     @property
     def _file_list_file(self) -> pathlib.Path:
         "Path to the file that stores the list of files in the downloaded dataset."
-        return self._pydax_dir / 'files.list'
+        # We use self._pydax_dir_ instead of self._pydax_dir because we don't want to have the directory created if it
+        # hasn't been yet when is_downloaded is called
+        return self._pydax_dir_ / 'files.list'
```

Comment:
Was this a bug?

Reply:
I don't think this is a bug by itself. Seems to me the best way to fix the issue is to create another property that ends with `_` that doesn't create the directory structure.

Reply:
Is there a situation though where `_file_list_file` needs to call `_pydax_dir` over `_pydax_dir_`? 

In `Dataset.download`, by the time `_file_list_file` is called, `_pydax_dir` was already called earlier. In `Dataset.is_downloaded` we just need to check if `_file_list_file` exists so we don't need `_pydax_dir` to have been called.

Reply:
`is_downloaded` is where I think it should be called

Reply:
can you elaborate a little bit on why though?

Reply:
It's just that the directory shouldn't be created in `is_downloaded`. The directory shouldn't be created merely because `is_downloaded` returns False.

Reply:
> It's just that the directory shouldn't be created in is_downloaded. The directory shouldn't be created merely because is_downloaded returns False.

I think one of us is confused then, because I agree and that's the change I am proposing